### PR TITLE
Update for 4.2

### PR DESCRIPTION
**Pull request by emteknetnz**
composer.json mostly does not require changing
```
    "require": {
        "silverstripe/framework": "^4"
    },
```

### CODE REVIEW

**Review by maxime-rainville — APPROVED** [review]
Looks good

File: .travis.yml [hunk 1]
```diff
@@ -9,15 +12,13 @@ env:
...
     - php: 7.1
-      env: DB=MYSQL RECIPE_VERSION="~4.2.0" PHPUNIT_TEST=1
-    - php: 7.1
-      env: DB=MYSQL RECIPE_VERSION="4.3.x-dev" PHPUNIT_TEST=1
-    - php: 7.1
-      env: DB=PGSQL RECIPE_VERSION="4.4.x-dev" PHPUNIT_TEST=1
+      env: DB=MYSQL RECIPE_VERSION="4.5.x-dev" PHPUNIT_TEST=1
     - php: 7.2
-      env: DB=MYSQL RECIPE_VERSION="4.5.x-dev" PHPUNIT_TEST=1 PHPCS_TEST=1
+      env: DB=PGSQL RECIPE_VERSION="4.6.x-dev" PHPUNIT_TEST=1 PHPUNIT_COVERAGE_TEST=1
```
Comment (ScopeyNZ) [inline .travis.yml]: You've doubled up on coverage tests here.

Comment (maxime-rainville) [inline .travis.yml]:
```suggestion
      env: DB=PGSQL RECIPE_VERSION="4.5.x-dev" PHPUNIT_TEST=1 PHPUNIT_COVERAGE_TEST=1
```

File: .travis.yml [hunk 2]
```diff
@@ -9,15 +12,13 @@ env:
...
     - php: 7.1
-      env: DB=MYSQL RECIPE_VERSION="~4.2.0" PHPUNIT_TEST=1
-    - php: 7.1
-      env: DB=MYSQL RECIPE_VERSION="4.3.x-dev" PHPUNIT_TEST=1
-    - php: 7.1
-      env: DB=PGSQL RECIPE_VERSION="4.4.x-dev" PHPUNIT_TEST=1
+      env: DB=MYSQL RECIPE_VERSION="4.5.x-dev" PHPUNIT_TEST=1
     - php: 7.2
-      env: DB=MYSQL RECIPE_VERSION="4.5.x-dev" PHPUNIT_TEST=1 PHPCS_TEST=1
+      env: DB=PGSQL RECIPE_VERSION="4.5.x-dev" PHPUNIT_TEST=1 PHPUNIT_COVERAGE_TEST=1
```
Comment (maxime-rainville) [inline .travis.yml]:
```suggestion
      env: DB=PGSQL RECIPE_VERSION="4.5.x-dev" PHPUNIT_TEST=1
```

**Pull request merged**